runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -351,7 +351,21 @@ public void onChatMessage(ChatMessage event)
 
 		if (config.screenshotKills() && KILL_MESSAGES.stream().anyMatch(chatMessage::contains))
 		{
-			String fileName = "Kill " + format(new Date());
+			String enemyName = "";
+			for (int i = 0; i <= KILL_MESSAGES.size(); i++)
+			{
+				if (chatMessage.contains(KILL_MESSAGES(i)))
+				{
+					enemyName = chatMessage.replace((String.valueOf(KILL_MESSAGES(i))),"");
+					break;
+				}
+				else 
+				{
+					i++;
+				}
+			}
+			if (enemyName)
+			String fileName = "Kill " + format(new Date()) + enemyName);
```

Comment:
This will be null if enemy name returns false. I would declare the file name where it was before and append if enemy name returns true.

Reply:
Are you even testing these? This will include the punctuation in the filename and it's the complete wrong way to do it. The KILL_MESSAGES should be converted to a regex with a group matcher for the name